cmd/guacone/cmd/query_vulnerability.go

```diff
@@ -105,72 +106,207 @@ var queryVulnCmd = &cobra.Command{
 				}
 			} else {
-				logger.Fatalf("failed to identify vulnerability as cve or ghsa and no results found for OSV")
+				logger.Debugf("failed to identify vulnerability as cve or ghsa and no results found for OSV")
 			}
-		}
-
-		var path []string
-		if osvResponse != nil {
-			path, err = queryCertifyVulnsNeighbors(ctx, gqlclient, pkgResponse, osvResponse.Osv[0].Id, model.EdgeOsvCertifyVuln, opts.depth)
-			if err != nil {
-				logger.Fatalf("error querying neighbor: %v", err)
+			if len(path) > 0 {
+				logger.Infof("found path %v", strings.Join(path, `,`))
+				logger.Infof("Visualizer url: http://localhost:3000/visualize?path=[%v]", strings.Join(path, `,`))
+			} else {
+				logger.Infof("No path to vulnerability ID found!")
 			}
-		} else if cveResponse != nil {
-			path, err = queryCertifyVulnsNeighbors(ctx, gqlclient, pkgResponse, cveResponse.Cve[0].Id, model.EdgeCveCertifyVuln, opts.depth)
+		} else {
+			var path []string
+			vulnPath, err := queryVulnsViaPackageNeighbors(ctx, gqlclient, pkgResponse.Packages[0].Namespaces[0].Names[0].Versions[0].Id, model.EdgePackageCertifyVuln)
 			if err != nil {
 				logger.Fatalf("error querying neighbor: %v", err)
 			}
-		} else if ghsaResponse != nil {
-			path, err = queryCertifyVulnsNeighbors(ctx, gqlclient, pkgResponse, ghsaResponse.Ghsa[0].Id, model.EdgeGhsaCertifyVuln, opts.depth)
+			depVulnPath, err := searchDependencyPackages(ctx, gqlclient, pkgResponse.Packages[0].Namespaces[0].Names[0].Versions[0].Id, opts.depth)
 			if err != nil {
-				logger.Fatalf("error querying neighbor: %v", err)
+				logger.Fatalf("error searching dependency packages match: %v", err)
+			}
+
+			path = append(path, vulnPath...)
+			path = append(path, depVulnPath...)
+
+			if len(path) > 0 {
+				path = append([]string{pkgResponse.Packages[0].Namespaces[0].Names[0].Versions[0].Id,
+					pkgResponse.Packages[0].Namespaces[0].Names[0].Id, pkgResponse.Packages[0].Namespaces[0].Id,
+					pkgResponse.Packages[0].Id}, path...)
+				logger.Infof("found path %v", strings.Join(path, `,`))
+				logger.Infof("Visualizer url: http://localhost:3000/visualize?path=[%v]", strings.Join(path, `,`))
+			} else {
+				logger.Infof("No path to vulnerabilities found!")
 			}
-		} else {
-			logger.Fatalf("failed to identify vulnerability as cve or ghsa and no results found for OSV")
 		}
-		logger.Infof("found path %v", strings.Join(path, `,`))
-		logger.Infof("Visualizer url: http://localhost:3000/visualize?path=[%v]", strings.Join(path, `,`))
 	},
 }
 
-func queryCertifyVulnsNeighbors(ctx context.Context, gqlclient graphql.Client, topPkgResponse *model.PackagesResponse, vulnerabilityNodeID string, edgeType model.Edge, depth int) ([]string, error) {
+func queryVulnsViaPackageNeighbors(ctx context.Context, gqlclient graphql.Client, pkgVersionID string, edgeType model.Edge) ([]string, error) {
+	path := []string{}
+	certifyVulns, err := model.Neighbors(ctx, gqlclient, pkgVersionID, []model.Edge{edgeType})
```

Comment:
> yea i was thinking more on the CLI side, where we check to see if the certifier has run (knowing if there is lack of data) so that the user doesn't get false negatives, but otherwise agreed.

Adding check for this.

Reply:
Added this check